### PR TITLE
chore: add some unit test on notification

### PR DESCRIPTION
**Pull request by acolombier**
As suggested in the issue #62 , I tasked myself with increasing test coverage and thought I will get started with notifications.

### CODE REVIEW

**Comment by acolombier** [comment]
Since this task requires some minimal refactoring in order to split the code in smaller independent sections, I thought I will start with `byte threshold` first and wait for your feedback before going ahead and doing all of them.

**Comment by GyulyVGC** [comment]
Sounds good. Just leave me the refactor tasks. The ideal thing is that you exclusively work inside "tests" modules.

**Comment by GyulyVGC** [comment]
However, if you already started refactoring keep going. 
Just let me know when and where you need it next time.

**Comment by GyulyVGC** [comment]
At a first read it seems a very good start btw! 
It's nice you decided to put that function in the notifications struct, feel free to do the same for the other notifications.
Thanks!

**Comment by acolombier** [comment]
Do you mind pointing out which bits your are referring to? Do you mean the `Default` trait implementation?

File: src/structs/notifications.rs [hunk 1]
```diff
@@ -44,6 +32,34 @@ pub struct PacketsNotification {
...
+            threshold: None,
+            sound: Sound::Gulp,
+            previous_threshold: 750,
+        }
+    }
+}
+
+impl PacketsNotification {
+    /// Arbitrary string constructor. Will fallback values to existing notification if set, default() otherwise
+    pub fn from(value: &str, existing: Option<Self>) -> Self {
```
Comment (GyulyVGC) [inline src/structs/notifications.rs:47]: This function content should be the same of the respective previous segment of code in settings.rs (e.g, when text field is empty, consider it as a zero and not as a None; I'd like this functions to have the same content and behaviour as the previous version of the code)

**Comment by acolombier** [comment]
I have reverted the `PacketsNotification` and `BytesNotification` behaviors. I kept the `trim` addition so there is more edge cases covered, although the UI doesn't let you input anything else.

**Comment by GyulyVGC** [comment]
Thanks! Once also the formatting will be fixed I'll review it! 👍

**Comment by GyulyVGC** [comment]
I just fixed the formatting (note that this can be automatically done with `cargo fmt`) and reviewed the PR and you did a great job.
It's nice you decided to start from the text input fields which are the most vulnerable part of the app.
Tyvm!

**Review by GyulyVGC — APPROVED** [review]

**Comment by GyulyVGC** [comment]
@all-contributors please add @acolombier for test translation

**Pull request merged**